src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -576,7 +589,32 @@ private static void EnsureNullableAttributeExists(CSharpCompilation compilation,
                         break;
 
                     case SyntaxKind.ReadOnlyKeyword:
-                        diagnostics.Add(ErrorCode.ERR_ReadOnlyNotSuppAsParamModDidYouMeanIn, modifier.GetLocation());
+                        Binder.CheckFeatureAvailability(modifier, MessageID.IDS_FeatureRefReadonlyParameters, diagnostics);
+
+                        if (seenReadonly)
+                        {
+                            addERR_DupParamMod(diagnostics, modifier);
+                        }
+                        else if (seenOut)
+                        {
+                            addERR_BadParameterModifiers(diagnostics, modifier, SyntaxKind.OutKeyword);
+                        }
+                        else if (seenIn)
+                        {
+                            addERR_BadParameterModifiers(diagnostics, modifier, SyntaxKind.InKeyword);
+                        }
+                        else if (seenParams)
+                        {
+                            addERR_ParamsCantBeWithModifier(diagnostics, modifier, SyntaxKind.ReadOnlyKeyword);
+                        }
+                        else if (!seenRef)
+                        {
+                            diagnostics.Add(ErrorCode.ERR_RefReadOnlyInverted, modifier);
+                        }
+                        else
+                        {
+                            seenReadonly = true;
+                        }
```

Comment:
It looks like `readonly` is allowed in combination with `this`. Can we add a test?

Reply:
Should we also add a test to double check that invoking `ref readonly` extension methods by pointers also works?
That is, this should work (just like it does for `ref` and `in` extension methods invoked by pointers):

```csharp
public static class Ex
{
    public static void Foo(this ref readonly int x)
    {
    }

    static void Test()
    {
        int x = 42;
        (&x)->Foo();
    }
}
```

Reply:
Thanks @Sergio0694. This PR is only about syntax, but will add a note to test plan.